iib/workers/tasks/utils.py

```diff
@@ -175,28 +176,49 @@ def _requires_max_ocp_version(bundle):
     return False
 
 
-def serve_index_registry(db_path):
+def serve_registry(base_dir, from_index=None, fbc_dir=None, db_path=None):
     """
     Locally start OPM registry service, which can be communicated with using gRPC queries.
 
     Due to IIB's paralellism, the service can run multiple times, which could lead to port
     binding conflicts. Resolution of port conflicts is handled in this function as well.
 
+    :param str base_dir: base directory to create temporary files in.
+    :param str from_index: index image to inspect.
+    :param str fbc_dir: path to file-based catalog directory.
     :param str db_path: path to index database containing the registry data.
     :return: tuple containing port number of the running service and the running Popen object.
     :rtype: (int, Popen)
     :raises IIBError: if all tried ports are in use, or the command failed for another reason.
     """
+    from iib.workers.tasks.build import _get_index_database
+
     conf = get_worker_config()
     port_start = conf['iib_grpc_start_port']
     port_end = port_start + conf['iib_grpc_max_port_tries']
 
+    assert (
```

Comment:
Could you please explain why is the assert use here? I am curious :)

Reply:
I am thinking about asserts as check for developers that they did not do anything that is not supported. In this case I wanted to be sure that nobody will use this function without any keyword argument or with multiple arguments set. Because it would not work otherwise.